lxd/devices.go

```diff
@@ -215,58 +226,20 @@ func deviceTaskBalance(d *Daemon) {
 			balancedContainers[c] = count
 		} else {
 			// Pinned
-			chunks := strings.Split(cpu, ",")
-			for _, chunk := range chunks {
-				if strings.Contains(chunk, "-") {
-					// Range
-					fields := strings.SplitN(chunk, "-", 2)
-					if len(fields) != 2 {
-						shared.Log.Error("Invalid limits.cpu value.", log.Ctx{"container": c.Name(), "value": cpu})
-						continue
-					}
-
-					low, err := strconv.Atoi(fields[0])
-					if err != nil {
-						shared.Log.Error("Invalid limits.cpu value.", log.Ctx{"container": c.Name(), "value": cpu})
-						continue
-					}
-
-					high, err := strconv.Atoi(fields[1])
-					if err != nil {
-						shared.Log.Error("Invalid limits.cpu value.", log.Ctx{"container": c.Name(), "value": cpu})
-						continue
-					}
+			cpus := parseCpuset(cpu)
```

Comment:
the error check is nice, but it would be better if we actually did get an err variable from parseCpuset :)